common/src/main/java/com/wynntils/features/user/players/PlayerArmorHidingFeature.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user.players;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureCategory;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.features.properties.StartDisabled;
+import com.wynntils.mc.event.PlayerArmorRenderEvent;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@StartDisabled
+@FeatureInfo(category = FeatureCategory.PLAYERS)
+public class PlayerArmorHidingFeature extends UserFeature {
+    @Config
+    public boolean hideHelmets = true;
+
+    @Config
+    public boolean hideChestplates = true;
+
+    @Config
+    public boolean hideLeggings = true;
+
+    @Config
+    public boolean hideBoots = true;
+
+    @Config
+    public boolean showCosmetics = true;
+
+    @SubscribeEvent
+    public void onPlayerArmorRender(PlayerArmorRenderEvent event) {
+        switch (event.getSlot()) {
+            case HEAD -> {
+                if (!hideHelmets) return;
+                if (!showCosmetics) { // helmet is hidden regardless, no extra logic needed
```

Comment:
Can you return early here too? Just a nit though

Reply:
where, exactly? this `if` statement does return, it just needs to cancel the event first

Reply:
if showcosmetics return;
e.cancel

Reply:
not every helmet is a cosmetic though - that logic wouldn't match the functionality

Reply:
oh, leave it then